portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -80,21 +85,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             e.printStackTrace();
         }
 
-        long timestamp = System.currentTimeMillis();
         String username = getAttribute(json, "username", "Anonymous");
         String comment = getAttribute(json, "comment", "...");
+        long timestamp = System.currentTimeMillis();
+        double sentiment = analyzeSentiment(comment);
+        String emotion =
+            sentiment >= -1 && sentiment <= -0.3 ? "Mad" :
+            sentiment > -0.3 && sentiment <= 0.3 ? "Neutral" :
+            "Happy";
```

Comment:
Nit: it seems like it might be beneficial to use an `Enum` for sentiment. Just something to consider, but the code is also totally okay as-is.